### ISSUE TITLE
rubocop prints  '=====..' before the formatted code

### ISSUE THREAD

**Issue by filipeisho** [rubocop-hq/rubocop#7938]
`rubocop -x -o /dev/null -s 'filename.rb'` prints unwanted stuff above the formatted file. `rubocop -x -o... ` should only print the formatted file.

--------

## Expected behavior

After running rubocop -x -o /dev/null -s 'filename.rb' I would expect rubocop to return my file formatted and nothing else.

## Actual behavior
Rubocop prints `====================` before the formatted file.

## Steps to reproduce the problem
 
rubocop -x -o /dev/null -s 'filename.rb'

## RuboCop version

```
$ [bundle exec] ~ rubocop -V
0.79.0 (using Parser 2.7.0.2, running on ruby 2.6.0 x86_64-darwin18)
```

**Comment by koic** [comment]
This `====================` line is the delimiter for auto-corrected source code and other stdout information.

```console
% cat example.rb
foo do
    bar
  end

% cat example.rb | rubocop -x -o /dev/null -s example.rb
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file:
 - Layout/SpaceAroundMethodCallOperator (0.82)
 - Lint/RaiseException (0.81)
 - Lint/StructNewOverride (0.81)
 - Style/ExponentialNotation (0.82)
 - Style/HashEachMethods (0.80)
 - Style/HashTransformKeys (0.80)
 - Style/HashTransformValues (0.80)
For more information: https://docs.rubocop.org/en/latest/versioning/
====================
foo do
  bar
end
```

The intent remained in the source code comment.

```ruby
# If we are asked to autocorrect source code read from stdin, the only
# reasonable place to write it is to stdout
# Unfortunately, we also write other information to stdout
# So a delimiter is needed for tools to easily identify where the
# autocorrected source begins
```

https://github.com/rubocop-hq/rubocop/blob/v0.82.0/lib/rubocop/cli/command/execute_runner.rb#L63-L67

So, I think that this is an expected behaviour.

**Comment by filipeisho** [comment]
Oh I see. Thanks

**Comment by lenormf** [comment]
Two ways:

- print warnings on `stderr`, if any, so that scripts can just ignore that stream and rely on the return code for actual errors
- detect that no additional messages need to be printed (which is what's happening in the reproducer), and so don't print any delimiters
